xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3764,6 +3776,13 @@ bool CVideoPlayer::CloseStream(CCurrentStream& current, bool bWaitForBuffers)
   if(bWaitForBuffers)
     SetCaching(CACHESTATE_DONE);
 
+  if (STREAM_SOURCE_MASK(current.source) == STREAM_SOURCE_DEMUX)
+  {
+    if (!m_pDemuxer)
+      return false;
```

Comment:
I don't think this is the correct location for doing this. The name of the method "CloseStream" is misleading. Actually is closes the stream player not the demux stream. We still want to be able to close the stream player even if there is no demuxer.